src/main/java/org/jenkinsci/main/modules/sshd/SSHD.java

```diff
@@ -132,7 +137,11 @@ public synchronized void start() throws IOException, InterruptedException {
         LOGGER.fine("starting SSHD");
 
         stop();
-        sshd = SshServer.setUpDefaultServer();
+        sshd = ServerBuilder.builder()
+                .randomFactory(new SingletonRandomFactory(SecurityUtils.isBouncyCastleRegistered()
+                        ? BouncyCastleNonBlockingRandomFactory.INSTANCE
+                        : JceRandomFactory.INSTANCE))
```

Comment:
We could also create a non-blocking version of JceRandomFactory, but I think that the BouncyCastle API plugins is so common that it should never be called. Thoughts?

Reply:
Works for me, a TODO comment would be useful